scripts/globals/effects/weakness.lua

```diff
@@ -3,47 +3,36 @@
 -- Weakness
 --
 -----------------------------------
+require("scripts/globals/status")
 
-require("scripts/globals/status");
-
------------------------------------
--- onEffectGain Action
------------------------------------
-
-function onEffectGain(target,effect)
+function onEffectGain(target, effect)
     --reduce HP and MP by the power amount. Add 100% slow
     --NOTE: The power amount dictates the amount to REDUCE MAX VALUES BY. E.g. Power=75 means 'reduce max hp/mp by 75%'
-    target:addMod(dsp.mod.HPP,-75);
-    target:addMod(dsp.mod.MPP,-75);
+    target:addMod(dsp.mod.HPP, -75)
+    target:addMod(dsp.mod.MPP, -75)
 
-    -- This is wrong, and players should be given a fixed 100% slow in the core so that all haste is ignored, but I am le tired.
-    target:addMod(dsp.mod.HASTE_MAGIC,-1024);
+    -- 100% Slow - FIXME: This kinda doesn't work since magic haste is clamped [-43.75%, 43.75%]
```

Comment:
should be clamped [-100, 43.75] probably

Reply:
You're saying that Slow from magic goes to -100%?

Reply:
don't know, it's either that or weakness slow is some special uncapped haste - do we know either way for sure?

Reply:
no, I don't think we do. But I think it's safe to clamp it further negative just because there aren't enough magic slow sources to push you that far without weakness.